2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh

```diff
@@ -384,20 +385,17 @@ kafka_configure_from_environment_variables() {
 #   None
 #########################
 kafka_initialize() {
-    # Since we remove this directory afterwards, it allows us to check if Kafka had already been initialized
-    if [[ ! -d "$KAFKA_BASEDIR"/configtmp ]]; then
-        info "Kafka has already been initialized"
-        return
-    fi
-
     info "Initializing Kafka..."
-
+    # DEPRECATED. Check for server.properties in old conf location to maintain compatibility with Helm chart.
+    if [[ -f "$KAFKA_BASEDIR"/conf/server.properties ]]; then
+      cp -L "$KAFKA_BASEDIR"/conf/server.properties "$KAFKA_CONF_FILE"
+    fi
     # Check for mounted configuration files
-    cp -r "$KAFKA_BASEDIR"/config/. "$KAFKA_CONFDIR"
-    rm -rf "$KAFKA_BASEDIR"/config
-    if [[ ! -f "$KAFKA_CONF_FILE" ]]; then
+    if ! is_dir_empty "$KAFKA_MOUNTED_CONFDIR"; then
+        cp -Lr "$KAFKA_MOUNTED_CONFDIR"/* "$KAFKA_CONFDIR"
+    fi
+    if [[ ! -f "$KAFKA_MOUNTED_CONFDIR"/server.properties ]] && [[ ! -f "$KAFKA_BASEDIR"/conf/server.properties ]]; then
```

Comment:
I think this change would make sense, since the configuration was already copied.

```suggestion
    if [[ ! -f "$KAFKA_BASEDIR"/conf/server.properties ]]; then
```

Reply:
No matter what happens there will always be a server.properties in $KAFKA_CONFDIR because of the changes to postunpack.sh, so we must check if the server.properties file was copied from either the old location or the new location to know if it is necessary to generate the default server.properties file.